samples/notebooks/prediction/SDK_Custom_Predict.ipynb

```diff
@@ -0,0 +1,1269 @@
+        "    -e AIP_PREDICT_ROUTE=/predict \\\n",
+        "    -e AIP_STORAGE_URI={BUCKET_NAME}/{MODEL_ARTIFACT_DIR} \\\n",
+        "    -e GOOGLE_APPLICATION_CREDENTIALS=credentials.json \\\n",
+        "    {REGION}-docker.pkg.dev/{PROJECT_ID}/{REPOSITORY}/{IMAGE}"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ce629eea32fd"
+      },
+      "outputs": [],
+      "source": [
+        "!curl localhost/health"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "56986f93438e"
+      },
+      "outputs": [],
+      "source": [
+        "!curl -X POST \\\n",
+        "  -d @instances.json \\\n",
+        "  -H \"Content-Type: application/json; charset=utf-8\" \\\n",
+        "  localhost/predict"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "a29fcbbe0188"
+      },
+      "outputs": [],
+      "source": [
+        "!docker stop local-iris"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "212b2935ea12"
+      },
+      "source": [
+        "### Push the container to artifact registry\n",
+        "\n",
+        "Configure Docker to access Artifact Registry. Then push your container image to your Artifact Registry repository."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wSFXCj3LdluJ"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud services list"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ABE9UpwSdluK"
+      },
+      "source": [
+        "If `artifactregistry.googleapis.com` is not enabled in your project, enable the API before proceeding."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "qDhLoQMydluK"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud services enable artifactregistry.googleapis.com"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "09ffe2434e3d"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud beta artifacts repositories create {REPOSITORY} \\\n",
+        "    --repository-format=docker \\\n",
+        "    --location=$REGION"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "293437024749"
+      },
+      "outputs": [],
+      "source": [
+        "!gcloud auth configure-docker {REGION}-docker.pkg.dev --quiet"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1dd7448f4703"
+      },
+      "outputs": [],
+      "source": [
+        "!docker push {REGION}-docker.pkg.dev/{PROJECT_ID}/{REPOSITORY}/{IMAGE}"
```

Comment:
We need to create the repository before we push if it's a new repo.

Reply:
For AR specifically.

Reply:
The repo is created a couple of lines above.